lib/sacastats_web/controllers/character_controller.ex

```diff
@@ -125,6 +137,275 @@ defmodule SacaStatsWeb.CharacterController do
     render(conn, "template.html", assigns)
   end
 
+  def reduce_stats(info) do
+    # MAY NEED LATER, NOT SURE
+    stat_by_faction_map =
+      info["stats"]["stat_by_faction"]
+      |> Enum.reduce(%{}, fn stat, acc ->
+        if stat["stat_name"] in ["kills", "killed_by"] do
+          Map.update(
+            acc,
+            stat["stat_name"],
+            %{
+              "value_forever_vs" => maybe_to_int(stat["value_forever_vs"]),
+              "value_forever_nc" => maybe_to_int(stat["value_forever_nc"]),
+              "value_forever_tr" => maybe_to_int(stat["value_forever_tr"])
+            },
+            fn map ->
+              map =
+                Map.update(
+                  map,
+                  "value_forever_vs",
+                  0,
+                  &(&1 + maybe_to_int(stat["value_forever_vs"]))
+                )
+
+              map =
+                Map.update(
+                  map,
+                  "value_forever_nc",
+                  0,
+                  &(&1 + maybe_to_int(stat["value_forever_nc"]))
+                )
+
+              Map.update(
+                map,
+                "value_forever_tr",
+                maybe_to_int(stat["value_forever_tr"]),
+                &(&1 + maybe_to_int(stat["value_forever_tr"]))
+              )
```

Comment:
Would pipes be better here?
```suggestion
              map 
              |> Map.update(
                "value_forever_vs",
                0,
                &(&1 + maybe_to_int(stat["value_forever_vs"]))
              )
              |> Map.update(
                "value_forever_nc",
                0,
                &(&1 + maybe_to_int(stat["value_forever_nc"]))
              )
              |> Map.update(
                "value_forever_tr",
                maybe_to_int(stat["value_forever_tr"]),
                &(&1 + maybe_to_int(stat["value_forever_tr"]))
              )
```